admin/statistics/class-statistics-service.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * @package WPSEO\Admin\Statistics
+ */
+
+/**
+ * Class WPSEO_Statistics_Service
+ */
+class WPSEO_Statistics_Service {
+
+	const CACHE_TRANSIENT_KEY = 'wpseo-dashboard-totals';
+
+	/**
+	 * @var WPSEO_Statistics
+	 */
+	protected $statistics;
+
+	/**
+	 * @var string[]
+	 */
+	protected $labels;
+
+	/**
+	 * WPSEO_Statistics_Service contructor.
+	 *
+	 * @param WPSEO_Statistics $statistics The statistics class to retrieve statistics from.
+	 */
+	public function __construct( WPSEO_Statistics $statistics = null ) {
+		if ( null === $statistics ) {
+			$statistics = new WPSEO_Statistics();
+		}
+
+		$this->statistics = $statistics;
+	}
+
+	/**
+	 * Fetches statistics by REST request.
+	 *
+	 * @return WP_REST_Response The response object.
+	 */
+	public function get_statistics() {
+		$statistics = $this->statistic_items();
+
+		$header = __( 'Below are your published posts&#8217; SEO scores. Now is as good a time as any to start improving some of your posts!', 'wordpress-seo' );
+
+		$data = array(
+			'header'     => $header,
+			'seo_scores' => $statistics,
+		);
+
+		return new WP_REST_Response( $data );
+	}
+
+	/**
+	 * An array representing items to be added to the At a Glance dashboard widget
+	 *
+	 * @return array
+	 */
+	private function statistic_items() {
+		$transient = get_transient( self::CACHE_TRANSIENT_KEY );
+		$user_id   = get_current_user_id();
+
+		if ( isset( $transient[ $user_id ] ) ) {
+			return $transient[ $user_id ];
+		}
+
+		return $this->set_statistic_items_for_this_user( $transient );
+	}
+
+	/**
+	 * Set the cache for a specific user
+	 *
+	 * @param array|boolean $transient The current stored transient with the cached data.
+	 *
+	 * @return mixed
+	 */
+	private function set_statistic_items_for_this_user( $transient ) {
+		if ( $transient === false ) {
+			$transient = array();
+		}
+
+		$user_id               = get_current_user_id();
```

Comment:
You have this value in the `statistics_items` where you are calling this method. I might be nicer to pass that variable as a dependency to this method.